pkg/platform/runtime/analytics/analytics.go

```diff
@@ -4,15 +4,15 @@ import (
 	"github.com/ActiveState/cli/internal/analytics"
 	anaConsts "github.com/ActiveState/cli/internal/analytics/constants"
 	"github.com/ActiveState/cli/internal/analytics/dimensions"
-	"github.com/ActiveState/cli/internal/constants"
 	"github.com/ActiveState/cli/internal/osutils"
+	"github.com/ActiveState/cli/pkg/platform/runtime/executors"
 )
 
 var isExecutor bool
 
 func init() {
-	if osutils.Executable() == constants.StateExecutorCmd {
-		isExecutor = true
+	if isExec, err := executors.IsExecutor(osutils.Executable()); err == nil {
```

Comment:
As I said executors do not use the analytics package directly, so I don't think we need this?

Reply:
That's right, but this code is checking the current executable. There is never a scenario in which an executor invokes the analytics code (whether sync or async) directly. Instead they talk directly to the svc without the analytics library. The svc then uses its analytics library to report the event.

In other words; I don't believe there is a scenario in which this conditional evaluates to true.

Reply:
Okay, you're right. I went through a debugger and that code does not appear to be hit. I've removed it.